### ISSUE TITLE
[statisticsPen] Unexpected values for meanX

### ISSUE THREAD

**Issue by n8willis** [fonttools/fonttools#2401]
I was working with statisticsPen some more and encountered some results from the `meanX` property that don't make sense to me.

Long story short; I expected that splitting a TTGlyph at meanX would give equal areas on both sides, and it doesn't.

Test case: a highly asymmetrical glyph like "r". Using Aclonica Regular from Google Fonts (it's alphabetically first), but it seems repeatable so far.
```
spen = StatisticsPen(glyphset)
glyphset["r"].draw(spen)
spen.meanX
``` 
= 444.9177309220464

Using some pathops & utility funcs to double-check that:
```
def clip_glyph(glyphName, glyphset, xMin, yMin, xMax, yMax):
    gpath = removeOverlaps.skPathFromGlyph(glyphName, glyphset)
    # make a Path from the clipping-box coordinates
    box = pathops.Path()
    box.moveTo(xMin, yMin)
    box.lineTo(xMax, yMin)
    box.lineTo(xMax, yMax)
    box.lineTo(xMin, yMax)
    box.close()
    # intersect the Paths
    clipped = pathops.op(gpath, box, pathops.PathOp.INTERSECTION)
    return removeOverlaps.ttfGlyphFromSkPath(clipped)

def area_metric(glyphName, glyphset, xMin, yMin, xMax, yMax):
    apen = AreaPen(glyphset)
    clipped_glyph = clip_glyph(glyphName, glyphset, xMin, yMin, xMax, yMax)
    clipped_glyph.draw(apen, glyphset)
    return apen.value
```
We don't get the areas I expected:
```
area_metric("r", glyphset, xMin, yMin, 444.9177309220464, yMax)
```
= 324184.0

```
area_metric("r", glyphset, 444.9177309220464, yMin, xMax, yMax)
```
= 172416.8333333333

Those are substantially apart. 
I bumped into this when making a binary search on that area_metric because I expected it to return the same as meanX, or at least within tolerance. Am I misunderstanding the meaning of `meanX`? Or doing something wrong in pathops?

**Comment by behdad** [comment]
My guess would be that the glyphs have overlaps, and that affects the area calculations. In your alternate codepath you are removing overlaps...

**Comment by behdad** [comment]
Try removing overlap then calling momentsPen, see if that fixes it.

**Comment by behdad** [comment]
> Unless you have drawn the original glyph, in which case I'm out of ideas.

Humm. Right. I have no idea. Let me look into it more.

**Comment by behdad** [comment]
What you have in mind is the median. Is hard to calculate via calculus.

**Comment by behdad** [comment]
https://www.diffen.com/difference/Mean_vs_Median

**Comment by n8willis** [comment]
Yeah, this is a pretty old thing that's a bit more tangled than I initially let on. I've been meaning to revisit it. I do still think there's an issue, but I believe it's somewhere else and I need to find time to write a more accurate report.

Short version is, [1] I actually first ran into problematic results in the higher moments (in this pen, I mean, not in the expanded version I was generating), then I tried to bring the test case down into the simpler moments because in theory that ought to be easier to test, then got confused for a bit. Doesn't help that [2] physicists and mechanical engineers have different notations for some of this stuff.

But yeah, not a bug here.
